hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/push/DelayedEventBusPushStrategy.java

```diff
@@ -57,26 +57,32 @@
     private static final Logger LOG = LoggerFactory.getLogger(DelayedEventBusPushStrategy.class);
 
     private static final int BLOCK_SIZE = 10_000;
-    private final ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+    private final ScheduledExecutorService executorService;
     private final BlockingDeque<org.eclipse.hawkbit.eventbus.event.Event> queue = new LinkedBlockingDeque<>(BLOCK_SIZE);
     private final EventBus.SessionEventBus eventBus;
     private final com.google.common.eventbus.EventBus systemEventBus;
+    private int uiid = -1;
```

Comment:
The use of `uiid` is not thread-safe. I suppose this was intended so `DispatchRunnable` is finished with the id it was  started with. Maybe you can use `vaadinUI.getUIId()` as it is already a instance variable of `DispatchRunnable` ?